source/linux/Foundational_Components/Kernel/Kernel_Drivers/DM-Timer.rst

```diff
@@ -0,0 +1,152 @@
+  Each timer includes compare logic to allow an interrupt event on a programmable counter
+  matching value. A dedicated output signal can be pulsed/toggled on an overflow or a match
+  event. This offers time-stamp trigger signaling or PWM signal sources. A dedicated input
+  signal can be used to trigger an automatic timer counter capture or an interrupt event on a
+  programmable input signal transition.
+
+- Each timer comes with pins that can be used as Timer Capture interrupts
+  or as timer PWM outputs.
+
+- For more detailed information on timers, please refer to the:
+  Peripherals/Timer Modules/Timers section of the device specific TRM.
+
+|
+
+**************
+Driver Sources
+**************
+
+- Driver source location: :file:`drivers/clocksource/ti,timer-dm.c`
+
+- Binding source location: :file:`Documentation/device-tree/bindings/timer/ti,timer-dm.yaml`
+
+PWM mode:
+
+- Driver source location: :file:`drivers/pwm/pwm-omap-dmtimer.c`
+
+- Binding source location: :file:`Documentation/devicetree/bindings/pwm/pwm-omap-dmtimer.txt`
+
+|
+
+********************
+Kernel Configuration
+********************
+
+Configs to be enabled in kernel:
+
+.. code-block:: menuconfig
+
+   Device Drivers --->
+      -> Clock Source drivers
+         -> OMAP dual-mode timer driver (OMAP_DM_TIMER [=y])
+
+PWM mode:
+
+.. code-block:: menuconfig
+
+   Device Drivers --->
+      Pulse-Width Modulation (PWM) Support (PWM [=y])
+         -> OMAP Dual-Mode Timer PWM support (PWM_OMAP_DMTIMER [=m])
+
+.. rubric:: Device tree configuration example
+
+In this example, we take am62x SK DT node as an example. In general all the timer nodes
+should be described in its appropriate domain .dtsi file, for example the following node
+can be found in k3-am62-main.dtsi:
+
+.. code-block:: dts
+
+	main_timer3: timer@2430000 {
+		compatible = "ti,am654-timer";
+		reg = <0x00 0x2430000 0x00 0x400>;
+		interrupts = <GIC_SPI 123 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&k3_clks 39 2>;
+		clock-names = "fck";
+		assigned-clocks = <&k3_clks 39 2>;
+		assigned-clock-parents = <&k3_clks 39 3>;
+		power-domains = <&k3_pds 39 TI_SCI_PD_EXCLUSIVE>;
+		ti,timer-pwm;
+	};
+
+PWM mode:
+
+The "ti,timer-pwm" DT attribute is required in the timer node in order to use this timer instance
+in PWM mode, if the timer supports PWM mode.
+
+An additional DT node is required to enable timer in PWM mode as shown below. The
+main_timer3_pins_default is also required and not listed here, use sysconfig tool for pinumux
+configuration.
+
+.. code-block:: dts
+
+	main_pwm3: dmtimer-main-pwm-3 {
+		compatible = "ti,omap-dmtimer-pwm";
+		#pwm-cells = <3>;
+		ti,timers = <&main_timer3>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&main_timer3_pins_default>;
+	};
+
+|
+
+************
+Driver Usage
+************
+
+For information on using DM timer in non-PWM mode, please refer to the detailed
+description in the TRM :file:`Peripherals/Timer Modules/Timers/Timers Functional Description`
```

Comment:
Are you referencing section depths as a file here? Seems a little intuitive. You could just say "see the Functional Description under the Timer Modules section"